src/frontend/src/optimizer/mod.rs

```diff
@@ -521,6 +521,13 @@ impl PlanRoot {
             ).into());
         }
 
+        if RwTimestampValidator::select_rw_timestamp_in_stream_query(plan.clone()) {
+            return Err(ErrorCode::NotSupported(
+                "selecting `_rw_timestamp` in a streaming query is not allowed".to_string(),
+                "please run the sql in batch mode or remove the column `_rw_timestamp` from the streaming query".to_string(),
```

Comment:
> and the current epoch for incremental data.

Yes, it is theoretically possible. We need to fetch the old epoch for delate mesage from the storage as well. Whethr we plan to support it in the future is depended on the requirement. If just for debugging purpose, I think batch query is enough, and it won't be invasive to the streaming proto.

Reply:
I think the problem here is the same as when we try to specify the `proctime` in the streaming query, which generates inconsistent delete messages. We don't know what the previous time value was when processing the delete message

Reply:
> We don't know what the previous time value was when processing the delete message

Makes sense to me. Some ideas:

- For append-only table / materialized views, there's no confusion and it sounds feasible.
- For table with on-conflict checks, since we already have to lookup the storage for every deletes, we can obtain the deleted timestamp at the same time.